src/meta/prom.rs

```diff
@@ -32,23 +40,56 @@ pub struct ClusterLeader {
     pub last_received: i64,
 }
 
-impl Metric {
-    pub fn new(
-        name: String,
-        value: f64,
-        collection: AHashMap<String, String>,
-        _timestamp: i64,
-        metric_type: String,
-    ) -> Self {
-        Metric {
-            name,
-            value,
-            collection,
-            metric_type,
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum MetricType {
+    UNKNOWN,
+    COUNTER,
+    GAUGE,
+    HISTOGRAM,
+    GAUGEHISTOGRAM,
+    SUMMARY,
+    INFO,
+    STATESET,
+}
```

Comment:
```suggestion
// Cf. https://github.com/prometheus/prometheus/blob/f5fcaa3872ce03808567fabc56afc9cf61c732cb/model/textparse/interface.go#L106-L119
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "UPPERCASE")]
pub enum MetricType {
    Counter,
    Gauge,
    Histogram,
    GaugeHistogram,
    Summary,
    Info,
    StateSet,
    Unknown,
}
```
```

Reply:
this define i copied from Prometheus, i think it is ok.